app/src/main/java/org/dhis2/usescases/datasets/dataSetTable/dataSetSection/DataSetIndicatorHolder.kt

```diff
@@ -0,0 +1,34 @@
+package org.dhis2.usescases.datasets.dataSetTable.dataSetSection
+
+import android.content.res.ColorStateList
+import android.view.View
+import android.widget.TextView
+import androidx.core.content.ContextCompat
+import androidx.core.view.ViewCompat
+import com.evrencoskun.tableview.adapter.recyclerview.holder.AbstractViewHolder
+import dhis2.org.R
+
+class DataSetIndicatorHolder(itemView: View) : AbstractViewHolder(itemView) {
+    fun bind(text: String, isHeader: Boolean) {
+        itemView.findViewById<TextView>(R.id.text).let {
+            it.text = text
+            if (isHeader) {
```

Comment:
I wouldn't add presentation logic to the holder, when you call this method in the _DataSetIndicatorAdapter_, either for the header (line 35) or the row (line 49) you have the data models, maybe the model could have Ui properties like `backgroundColor: Int` and `textcolor: Int`.
What do you think?

Reply:
I would agree with @andresmr in here, however I don't know if we have a data model in here. Looking at the methods in the adapter this "models" can be anything (Any), so don't know if we can a proper data model for this holder.

or am I wrong??

Reply:
Yes, you are right, they are set toString when calling binding, well I don't know is this would be out of the scope of the ticket, but we could think how these data models are created.